src/openzeppelin/token/erc20/dual20.cairo

```diff
@@ -0,0 +1,142 @@
+use array::ArrayTrait;
+use array::SpanTrait;
+use core::result::ResultTrait;
+use integer::Felt252TryIntoU8;
+use option::OptionTrait;
+use starknet::call_contract_syscall;
+use starknet::ContractAddress;
+use starknet::Felt252TryIntoContractAddress;
+use starknet::SyscallResultTrait;
+use traits::Into;
+use traits::TryInto;
+
+use openzeppelin::utils::try_selector_with_fallback;
+use openzeppelin::utils::Felt252TryIntoBool;
+use openzeppelin::utils::BoolIntoFelt252;
+use openzeppelin::utils::selectors;
+use openzeppelin::utils::serde::SerializedAppend;
+
+#[derive(Copy, Drop)]
+struct DualERC20 {
+    contract_address: ContractAddress
+}
+
+trait DualERC20Trait {
+    fn name(self: @DualERC20) -> felt252;
+    fn symbol(self: @DualERC20) -> felt252;
+    fn decimals(self: @DualERC20) -> u8;
+    fn total_supply(self: @DualERC20) -> u256;
+    fn balance_of(self: @DualERC20, account: ContractAddress) -> u256;
+    fn allowance(self: @DualERC20, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(self: @DualERC20, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        self: @DualERC20, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(self: @DualERC20, spender: ContractAddress, amount: u256) -> bool;
+}
+
+impl DualERC20Impl of DualERC20Trait {
+    fn name(self: @DualERC20) -> felt252 {
+        *call_contract_syscall(*self.contract_address, selectors::name, ArrayTrait::new().span())
+            .unwrap_syscall()
+            .at(0)
+    }
+
+    fn symbol(self: @DualERC20) -> felt252 {
+        *call_contract_syscall(*self.contract_address, selectors::symbol, ArrayTrait::new().span())
+            .unwrap_syscall()
+            .at(0)
+    }
+
+    fn decimals(self: @DualERC20) -> u8 {
+        (*call_contract_syscall(
+            *self.contract_address, selectors::decimals, ArrayTrait::new().span()
+        )
+            .unwrap_syscall()
+            .at(0))
+            .try_into()
+            .unwrap()
+    }
+
+    fn total_supply(self: @DualERC20) -> u256 {
+        let snake_selector = selectors::total_supply;
+        let camel_selector = selectors::totalSupply;
+
+        let mut args = ArrayTrait::new();
```

Comment:
We are using `ArrayTrait::new()` on the above method directly, while we are using args here. We should maintain consistency. I personally like more the empty args approach, because I think is easier to read, even when is an empty array.

Reply:
Mmm this is true. I agree that it's easier to read. Will update